tools/cmake/FindBpfObject.cmake

```diff
@@ -0,0 +1,184 @@
+    else()
+      message(FATAL_ERROR "Failed to parse clang version string: ${CLANG_version_output}")
+    endif()
+  else()
+    message(FATAL_ERROR "Command \"${BPFOBJECT_CLANG_EXE} --version\" failed with output:\n${CLANG_version_error}")
+  endif()
+endif()
+
+if(NOT LIBBPF_INCLUDE_DIRS OR NOT LIBBPF_LIBRARIES)
+  find_package(LibBpf)
+endif()
+
+if(BPFOBJECT_VMLINUX_H)
+  get_filename_component(GENERATED_VMLINUX_DIR ${BPFOBJECT_VMLINUX_H} DIRECTORY)
+elseif(BPFOBJECT_BPFTOOL_EXE)
+  # Generate vmlinux.h
+  set(GENERATED_VMLINUX_DIR ${CMAKE_CURRENT_BINARY_DIR})
+  set(BPFOBJECT_VMLINUX_H ${GENERATED_VMLINUX_DIR}/vmlinux.h)
+  execute_process(COMMAND ${BPFOBJECT_BPFTOOL_EXE} btf dump file /sys/kernel/btf/vmlinux format c
+    OUTPUT_FILE ${BPFOBJECT_VMLINUX_H}
+    ERROR_VARIABLE VMLINUX_error
+    RESULT_VARIABLE VMLINUX_result)
+  if(${VMLINUX_result} EQUAL 0)
+    set(VMLINUX ${BPFOBJECT_VMLINUX_H})
+  else()
+    message(FATAL_ERROR "Failed to dump vmlinux.h from BTF: ${VMLINUX_error}")
+  endif()
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(BpfObject
+  REQUIRED_VARS
+    BPFOBJECT_BPFTOOL_EXE
+    BPFOBJECT_CLANG_EXE
+    LIBBPF_INCLUDE_DIRS
+    LIBBPF_LIBRARIES
+    GENERATED_VMLINUX_DIR)
+
+# Get clang bpf system includes
+execute_process(
+  COMMAND bash -c "${BPFOBJECT_CLANG_EXE} -v -E - < /dev/null 2>&1 |
+          sed -n '/<...> search starts here:/,/End of search list./{ s| \\(/.*\\)|-idirafter \\1|p }'"
+  OUTPUT_VARIABLE CLANG_SYSTEM_INCLUDES_output
+  ERROR_VARIABLE CLANG_SYSTEM_INCLUDES_error
+  RESULT_VARIABLE CLANG_SYSTEM_INCLUDES_result
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+if(${CLANG_SYSTEM_INCLUDES_result} EQUAL 0)
+  string(REPLACE "\n" " " CLANG_SYSTEM_INCLUDES ${CLANG_SYSTEM_INCLUDES_output})
+  message(STATUS "BPF system include flags: ${CLANG_SYSTEM_INCLUDES}")
+else()
+  message(FATAL_ERROR "Failed to determine BPF system includes: ${CLANG_SYSTEM_INCLUDES_error}")
+endif()
+
+# Get target arch
+execute_process(COMMAND uname -m
+  COMMAND sed "s/x86_64/x86/"
+  OUTPUT_VARIABLE ARCH_output
+  ERROR_VARIABLE ARCH_error
+  RESULT_VARIABLE ARCH_result
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+if(${ARCH_result} EQUAL 0)
+  set(ARCH ${ARCH_output})
+  message(STATUS "BPF target arch: ${ARCH}")
+else()
+  message(FATAL_ERROR "Failed to determine target architecture: ${ARCH_error}")
+endif()
+
+# Public macro
+macro(bpf_object name input)
+  set(BPF_C_FILE ${CMAKE_CURRENT_SOURCE_DIR}/${input})
+  set(BPF_O_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.bpf.o)
+  set(BPF_SKEL_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.skel.h)
+  set(OUTPUT_TARGET ${name}_skel)
+
+  # Build BPF object file
+  add_custom_command(OUTPUT ${BPF_O_FILE}
+    COMMAND ${BPFOBJECT_CLANG_EXE} -g -O2 -target bpf -D__TARGET_ARCH_${ARCH}
+            ${CLANG_SYSTEM_INCLUDES} -I${GENERATED_VMLINUX_DIR}
+            -isystem ${LIBBPF_INCLUDE_DIRS} -c ${BPF_C_FILE} -o ${BPF_O_FILE}
```

Comment:
I added the `-isystem ${LIBBPF_INCLUDE_DIRS}` flag so that the vendored libbpf headers can be found by the .bpf.c. Not sure if this should be `-idirafter` but I think it should work regardless.

Reply:
I think -isystem is right, these libbpf headers should be preferred over system ones